contributor-docs/adrs/adr-005-box-sx.md

```diff
@@ -0,0 +1,62 @@
+# ADR 005: Use Box as a building block for all other components
+
+## Status
+
+Proposed
+
+## Context
+
+In Primer React and consuming applications, we use many different patterns for creating React components. Two common patterns are:
+
+1. Creating components with styled-components
+
+    ```tsx
+    const Avatar = styled.img.attrs<StyledAvatarProps>(props => ({
+      height: props.size,
+      width: props.size
+    }))<StyledAvatarProps>`
+      display: inline-block;
+      overflow: hidden;
+      line-height: ${get('lineHeights.condensedUltra')};
+      border-radius: ${props => getBorderRadius(props)};
+      ${sx}
+    `
+    ```
+
+    [Show full code example →](https://github.com/primer/react/pull/2019/files?diff=split&w=0)
+
+2.  Creating components with Box
+
+    ```tsx
+    const Avatar: React.FC<AvatarProps> = ({size = 20, alt = '', square = false, sx = {}, ...props}) => {
+      const styles = {
```

Comment:
should we avoid `styles` here, since it's very close to sounding like inline styles vs `localSx`? or something like that

should we add types here too?

```
const localSx: BetterStyledSystemObject (is this still the type to use) = {}
```

Reply:
`componentStyles` or `baseStyles` can be a better variable here. I don't want to prescriptive about the variable name in this ADR here because it depends on the context. (for example, Button has `baseStyles` + `variantStyles` + `sizeStyles`)

`BetterStyledSystemObject` is the correct type to use, however it could use some improvements as well. [Example sandbox](https://codesandbox.io/s/proud-glitter-q3z5rh?file=/src/Avatar.tsx:884-1309)

Reply:
Totally!  Just wanted to make sure we don't cause confusion with inline styles here, more than variable naming!